### DISCUSSION TITLE
Load_var to use in source

### DISCUSSION THREAD

**Discussion by luis-n4489** [concourse/concourse#7714] [Help & Support]
So docker-image-resource is currently not able to use IAM roles, BUT i was thinking of then using load_var for this case, i dont think its possible to pass the credentials to a source / resource is there?

**Comment by ringods** [comment]
Hello @luis-nagaki,

Are your referring to this resource type?

https://github.com/concourse/docker-image-resource

If so, I suggest to switch to this one instead:

https://github.com/concourse/registry-image-resource

And coming back to your question, do you mean an AWS IAM role associated to your worker? Concourse doesn't let worker information leak into the builds, by design. You will have to pass credentials explicitly to your jobs. We hook up Concourse to our Hashicorp Vault setup:

https://concourse-ci.org/vault-credential-manager.html

Vault is then configured with the cloud provider to create session credentials every time Concourse requests a secret for a build. 

Does my answer help you any further?

**Reply by luis-n4489** [reply]
@ringods yeah you actually refreshed my memory - and i totally forgot about that resource. It either does not work, I am assuming a role that worker already can do (using just a reg. task) but it still fails. OR there needs to be some more example / clarity into the resource. https://github.com/concourse/registry-image-resource/issues/277

hopefully it will nudge the thread to get seen
